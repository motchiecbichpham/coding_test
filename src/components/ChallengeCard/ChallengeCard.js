import React, {memo} from 'react';
import {View, Text, Image, StyleSheet} from 'react-native';
import Ionicon from 'react-native-vector-icons/Ionicons';
import {DEFAULT_ICON} from '../../utils/constant';
import StyledText from 'react-native-styled-text';
import {COLORS} from '../../utils/colors';

const ChallengeCard = memo(({props}) => {
  return (
    <View style={styles.container}>
      <Image style={styles.image} source={props.image} />
      <View style={styles.contentContainer}>
        <Text numberOfLines={3} style={styles.title}>
          {props.title}
        </Text>
        <Text numberOfLines={3} style={styles.subTitle}>
          {props.subTitle}
        </Text>
        <View style={styles.descriptionContainer}>
          <StyledText style={styles.description}>
            {props.description.replace('{{num}}', props.point || '0')}
          </StyledText>
          <Ionicon
            name={DEFAULT_ICON.name}
            color={DEFAULT_ICON.color}
            size={DEFAULT_ICON.size}
          />
        </View>
      </View>
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 8,
    paddingVertical: 16,
    borderRadius: 8,
    width: 150,
    height: 300,
    shadowColor: COLORS.black,
    elevation: 5,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.2,
    shadowRadius: 2,
    backgroundColor: COLORS.white,
    margin: 16,
  },
  image: {
    width: 100,
    maxHeight: 120,
    alignSelf: 'center',
  },
  title: { fontWeight: '700', width: '100%', color: COLORS.darkCharcoal },
  description: {
    width: '70%',
  },
  subTitle: {
    fontWeight: '300',
    color: COLORS.grey,
    marginTop: 8,
  },
  contentContainer: {
    marginTop: 16,
  },
  descriptionContainer: {
    marginTop: 32,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
});
export default ChallengeCard;
