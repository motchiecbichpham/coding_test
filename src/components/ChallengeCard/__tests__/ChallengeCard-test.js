import React from 'react';
import renderer from 'react-test-renderer';
import ChallengeCard from '../ChallengeCard';
const challengeProps = {
  id: 1,
  title: 'Advanced Health Screening',
  description: 'Earn up to <b>{{num}}</b> pts',
  image: require('../../../asset/challenge1.jpeg'),
  point: 99999,
};
const challengePropsWithoutPoint = {
  id: 1,
  title: 'Advanced Health Screening',
  description: 'Earn up to <b>{{num}}</b> pts',
  image: require('../../../asset/challenge1.jpeg'),
};
test('renders with point = 99999', () => {
  const tree = renderer
    .create(<ChallengeCard props={challengeProps} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders without point', () => {
  const tree = renderer
    .create(<ChallengeCard props={challengePropsWithoutPoint} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
