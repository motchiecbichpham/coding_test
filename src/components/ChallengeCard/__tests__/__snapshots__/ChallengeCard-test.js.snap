// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with point = 99999 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 5,
      "height": 300,
      "margin": 16,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
      "width": 150,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/asset/challenge1.jpeg",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "maxHeight": 120,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": "#333333",
          "fontWeight": "700",
          "width": "100%",
        }
      }
    >
      Advanced Health Screening
    </Text>
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": undefined,
          "fontWeight": "300",
          "marginTop": 8,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 32,
        }
      }
    >
      <Text
        style={
          Object {
            "width": "70%",
          }
        }
      >
        Earn up to 
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          99999
        </Text>
         pts
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#EB596E",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders without point 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 5,
      "height": 300,
      "margin": 16,
      "paddingHorizontal": 8,
      "paddingVertical": 16,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
      "width": 150,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/asset/challenge1.jpeg",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "maxHeight": 120,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": "#333333",
          "fontWeight": "700",
          "width": "100%",
        }
      }
    >
      Advanced Health Screening
    </Text>
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": undefined,
          "fontWeight": "300",
          "marginTop": 8,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 32,
        }
      }
    >
      <Text
        style={
          Object {
            "width": "70%",
          }
        }
      >
        Earn up to 
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          0
        </Text>
         pts
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#EB596E",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
